include/capstone/tricore.h

```diff
@@ -0,0 +1,582 @@
+  TriCore_INS_OR_GE,
+  TriCore_INS_OR_LT_U,
+  TriCore_INS_OR_LT,
+  TriCore_INS_OR_NE,
+  TriCore_INS_OR_NOR_T,
+  TriCore_INS_OR_OR_T,
+  TriCore_INS_OR_T,
+  TriCore_INS_OR,
+  TriCore_INS_PACK,
+  TriCore_INS_PARITY,
+  TriCore_INS_POPCNT_W,
+  TriCore_INS_Q31TOF,
+  TriCore_INS_QSEED_F,
+  TriCore_INS_RESTORE,
+  TriCore_INS_RET,
+  TriCore_INS_RFE,
+  TriCore_INS_RFM,
+  TriCore_INS_RSLCX,
+  TriCore_INS_RSTV,
+  TriCore_INS_RSUBS_U,
+  TriCore_INS_RSUBS,
+  TriCore_INS_RSUB,
+  TriCore_INS_SAT_BU,
+  TriCore_INS_SAT_B,
+  TriCore_INS_SAT_HU,
+  TriCore_INS_SAT_H,
+  TriCore_INS_SELN_A,
+  TriCore_INS_SELN,
+  TriCore_INS_SEL_A,
+  TriCore_INS_SEL,
+  TriCore_INS_SHAS,
+  TriCore_INS_SHA_B,
+  TriCore_INS_SHA_H,
+  TriCore_INS_SHA,
+  TriCore_INS_SHUFFLE,
+  TriCore_INS_SH_ANDN_T,
+  TriCore_INS_SH_AND_T,
+  TriCore_INS_SH_B,
+  TriCore_INS_SH_EQ,
+  TriCore_INS_SH_GE_U,
+  TriCore_INS_SH_GE,
+  TriCore_INS_SH_H,
+  TriCore_INS_SH_LT_U,
+  TriCore_INS_SH_LT,
+  TriCore_INS_SH_NAND_T,
+  TriCore_INS_SH_NE,
+  TriCore_INS_SH_NOR_T,
+  TriCore_INS_SH_ORN_T,
+  TriCore_INS_SH_OR_T,
+  TriCore_INS_SH_XNOR_T,
+  TriCore_INS_SH_XOR_T,
+  TriCore_INS_SH,
+  TriCore_INS_STLCX,
+  TriCore_INS_STUCX,
+  TriCore_INS_ST_A,
+  TriCore_INS_ST_B,
+  TriCore_INS_ST_DA,
+  TriCore_INS_ST_D,
+  TriCore_INS_ST_H,
+  TriCore_INS_ST_Q,
+  TriCore_INS_ST_T,
+  TriCore_INS_ST_W,
+  TriCore_INS_SUBC,
+  TriCore_INS_SUBSC_A,
+  TriCore_INS_SUBS_BU,
+  TriCore_INS_SUBS_B,
+  TriCore_INS_SUBS_HU,
+  TriCore_INS_SUBS_H,
+  TriCore_INS_SUBS_U,
+  TriCore_INS_SUBS,
+  TriCore_INS_SUBX,
+  TriCore_INS_SUB_A,
+  TriCore_INS_SUB_B,
+  TriCore_INS_SUB_F,
+  TriCore_INS_SUB_H,
+  TriCore_INS_SUB,
+  TriCore_INS_SVLCX,
+  TriCore_INS_SWAPMSK_W,
+  TriCore_INS_SWAP_A,
+  TriCore_INS_SWAP_W,
+  TriCore_INS_SYSCALL,
+  TriCore_INS_TLBDEMAP,
+  TriCore_INS_TLBFLUSH_A,
+  TriCore_INS_TLBFLUSH_B,
+  TriCore_INS_TLBMAP,
+  TriCore_INS_TLBPROBE_A,
+  TriCore_INS_TLBPROBE_I,
+  TriCore_INS_TRAPSV,
+  TriCore_INS_TRAPV,
+  TriCore_INS_UNPACK,
+  TriCore_INS_UPDFL,
+  TriCore_INS_UTOF,
+  TriCore_INS_WAIT,
+  TriCore_INS_XNOR_T,
+  TriCore_INS_XNOR,
+  TriCore_INS_XOR_EQ,
+  TriCore_INS_XOR_GE_U,
+  TriCore_INS_XOR_GE,
+  TriCore_INS_XOR_LT_U,
+  TriCore_INS_XOR_LT,
+  TriCore_INS_XOR_NE,
+  TriCore_INS_XOR,
+
+	// clang-format on
+	// generate content <TriCoreGenCSInsnEnum.inc> end
+	TriCore_INS_ENDING, // <-- mark the end of the list of instructions
+} tricore_insn;
+
+//> Group of TriCore instructions
+typedef enum tricore_insn_group {
+	TriCore_GRP_INVALID, ///< = CS_GRP_INVALID
+	//> Generic groups
+	TriCore_GRP_CALL, ///< = CS_GRP_CALL
+	TriCore_GRP_JUMP, ///< = CS_GRP_JUMP
+	TriCore_GRP_ENDING, ///< = mark the end of the list of groups
+} tricore_insn_group;
+
+typedef enum tricore_feature_t {
+
+	TriCore_FEATURE_INVALID = 0,
+	// generate content <TriCoreGenCSFeatureEnum.inc> begin
+	// clang-format off
+
+/* Capstone Disassembly Engine, https://www.capstone-engine.org */
```

Comment:
Should we also remove the header here? I also wonder if maybe it's better to also follow the indent here

Reply:
@kabeor @aquynh @Rot127  do you agree with me? I think it's the only thing left before it could be merged, right?